komodo-relational/src/main/java/org/komodo/relational/teiid/internal/TeiidImpl.java

```diff
@@ -1007,6 +942,244 @@ public CachedTeiid importContent(UnitOfWork transaction) throws KException {
+        if(cachedTeiid!=null) {
+            TeiidInstance teiidInstance = getConnectedTeiidInstance(transaction);
+
+            try {
+                TeiidTranslator instTranslator = teiidInstance.getTranslator(translatorName);
+                if (instTranslator == null) return;
+
+                addOrReplaceCachedTranslator(transaction, instTranslator, cachedTeiid);
+                
+            } catch (Exception ex) {
+                throw new KException(ex);
+            }            
+        }
+    }
+    
+    @Override
+    public void updateCacheWithServerDriver(UnitOfWork transaction, String driverName) throws KException {
+        ArgCheck.isNotNull( transaction, "transaction" ); //$NON-NLS-1$
+        ArgCheck.isTrue( ( transaction.getState() == State.NOT_STARTED ), "transaction state is not NOT_STARTED" ); //$NON-NLS-1$
+
+        // If the CachedTeiid exists, replace the Driver
+        CachedTeiid cachedTeiid = getCachedTeiid(transaction);
+        if(cachedTeiid!=null) {
+            TeiidInstance teiidInstance = getConnectedTeiidInstance(transaction);
+
+            try {
+                Collection<String> instDataSourceTypeNames = teiidInstance.getDataSourceTypeNames();
+                if(instDataSourceTypeNames.contains(driverName)) {
+                    addOrReplaceCachedDriver(transaction, driverName, cachedTeiid);
+                }
+            } catch (Exception ex) {
+                throw new KException(ex);
+            }            
+        }
+    }
+    
+    /*
+     * Get a connected teiid instance.  If there is a problem getting the instance or connecting - an exception is thrown.
+     */
+    private TeiidInstance getConnectedTeiidInstance(UnitOfWork transaction) throws KException {
+        TeiidInstance teiidInstance = getTeiidInstance(transaction);
+        if (teiidInstance == null) {
+            throw new KException(Messages.getString(Messages.Relational.TEIID_INSTANCE_ERROR));
+        }
+        try {
+            teiidInstance.connect();
+            if (! teiidInstance.isConnected()) {
+                throw new KException(Messages.getString(Messages.Relational.TEIID_INSTANCE_CONNECTION_ERROR));
+            }
+        } catch (Exception ex) {
+            throw new KException(ex);
+        }
+        return teiidInstance;
+    }
+    
+    /*
+     * Returns the CachedTeiid instance if it exists.  If doesnt exist, null is returned.
+     */
+    private CachedTeiid getCachedTeiid(UnitOfWork transaction) throws KException {
+        CachedTeiid cachedTeiid = null;
+        
+        KomodoObject teiidCache = getRepository().komodoTeiidCache(transaction);
+        final String id = getName(transaction);
+        if (teiidCache.hasChild(transaction, id)) {
+            KomodoObject child = teiidCache.getChild(transaction, id);
+            cachedTeiid = CachedTeiid.RESOLVER.resolve(transaction, child);
+        } 
+        
+        return cachedTeiid;
```

Comment:
The expectation for this method should be that it always returns a CachedTeiid object. What we are asking is to get the cache version of this Teiid instance. Thus, should it not try to importContent() to create a cache and only then return null if it fails?